src/components/crossForm/CrossForm.vue

```diff
@@ -230,18 +238,16 @@ import { Field, Form, ErrorMessage } from 'vee-validate'
 import BigNumber from 'bignumber.js'
```

Comment:
@pmprete doubt: why use this instead of [web3-utils's BN](https://web3js.readthedocs.io/en/v1.2.11/web3-utils.html#utils-bn)? (no idea, just curiosity)

Reply:
maybe because [it supports more number formats](https://dev.to/fvictorio/a-comparison-of-bignumber-libraries-in-javascript-2gc5)?

Reply:
BN is number **without** decimals, so convert it to decimal you will need to convert it to string, then use [fromWei](https://web3js.readthedocs.io/en/v1.2.11/web3-utils.html?highlight=toWei#fromwei), but you got a string, so you need to convert it to number, and loose precision when doing this.
Also when using fromWei you need to pass the unit as a string, and we have decimals as a number.
Native js BigNumber does not cut it either as it's very limited
Ethers.js BigNumber is broken and does no accept values bigger than MAX_INT
So, in my experience bignumber.js is currently the best library to handle them